idea/src/org/jetbrains/kotlin/idea/inspections/collections/ConvertCallChainIntoSequenceInspection.kt

```diff
@@ -130,6 +130,7 @@ private fun KtQualifiedExpression.collectCallExpression(context: BindingContext)
         .dropWhile { !it.isTransformationOrTermination(context) }
         .takeWhile { it.isTransformationOrTermination(context) && !it.hasReturn() }
         .toList()
+        .dropLastWhile { it.calleeExpression?.text == "groupingBy" }
```

Comment:
Why you did not want just to remove `groupingBy` from termination list, is there some reason I do not see?

Reply:
Ok, I will add this as an additional test and probably create `listOf("groupingBy")` as a list of some "lazy terminators".